src/selective-reader.test.js

```diff
@@ -0,0 +1,100 @@
+//
+// Copyright 2019 DXOS.org
+//
+
+import pify from 'pify';
+import ram from 'random-access-memory';
+import waitForExpect from 'wait-for-expect';
+import { FeedStore } from './feed-store';
+
+function append (feed, message) {
+  return pify(feed.append.bind(feed))(message);
+}
+
+async function generateStreamData (feedStore, maxMessages = 200) {
+  const [feed1, feed2] = await Promise.all([
+    feedStore.openFeed('/feed1'),
+    feedStore.openFeed('/feed2')
+  ]);
+
+  const messages = [];
+  for (let i = 0; i < maxMessages; i++) {
+    messages.push(append(feed1, `feed1/message${i}`));
+    messages.push(append(feed2, `feed2/message${i}`));
+  }
+
+  await Promise.all(messages);
+
+  return [feed1, feed2];
+}
+
+describe('SelectiveReader', () => {
+  test('two feeds', async () => {
+    const feedStore = await FeedStore.create(ram, { feedOptions: { valueEncoding: 'utf-8' } });
```

Comment:
Consider testing with persistent storage since many bugs and performance problems seen in the past with this kind of code vs ram storage.